lib/glific/contacts.ex

```diff
@@ -275,9 +275,22 @@ defmodule Glific.Contacts do
   end
 
   @doc """
-  Check if this contact id is a new conatct
+  This function is called by the messaging framework for all incoming messages, hence
+  might be a good candidate to maintain a contact level cache at some point
+
+  We use a fetch followed by create, to avoid the explosion in the id namespace
   """
+  @spec maybe_create_contact(map()) :: {:ok, Contact.t()} | {:error, Ecto.Changeset.t()}
+  def maybe_create_contact(sender) do
+    case Repo.get_by(Contact, %{phone: sender.phone}) do
```

Comment:
Any specific reason to not use fetch by?

Reply:
since we are fine with nil in this case, since we are resolving the error, so did not need an ok code here